### PR TITLE
[W5.4][F10-A4] Wen Xin

### PR DESCRIPTION
**Pull request by wenmogu**
restore the most recent deletion

### CODE REVIEW

**Comment by wenmogu** [comment]
Emmmm sorry i dont really get it...  the Travis CI job log shows there are "differences" between the "expected" and "actual" (i dont know what files they are), but in my case, the "differences" are actually the same, meaning the "expected" and the "actual" are the same...so what is happening?

File: src/seedu/addressbook/commands/DeleteCommand.java [hunk 1]
```diff
@@ -27,15 +28,20 @@ public DeleteCommand(int targetVisibleIndex) {
 
     @Override
     public CommandResult execute() {
+        //before removal of the person, put this person into the storage of the removed
         try {
             final ReadOnlyPerson target = getTargetPerson();
+            storageForDeleted.storeTheDeleted(target);
+            // System.out.println("i m after the storeTheDeleted");
```
Comment (whipermr5) [inline src/seedu/addressbook/commands/DeleteCommand.java]: Avoid commented out code.

File: src/seedu/addressbook/commands/RestoreCommand.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    public static final String COMMAND_WORD = "restore";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD
+            + ": Revoke the most recent deletion\n"
+            + "Example: " + COMMAND_WORD;
+
+
+    @Override
+    public CommandResult execute() {
+        // System.out.println("I m seeing if the storage works: " + storageForDeleted.peek().toString());
```
Comment (whipermr5) [inline src/seedu/addressbook/commands/RestoreCommand.java]: Same here.

File: src/seedu/addressbook/commands/RestoreCommand.java [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+        try {
+            Person toAdd = storageForDeleted.getPersonToBeRestored();
+            if (toAdd == null) {
+                throw new UnsupportedOperationException("The person to be added is null");
+            } else {
+                addressBook.addPerson(toAdd);
+                return new CommandResult(String.format(MESSAGE_SUCCESS, toAdd));
+            }
+        } catch (Exception e) {
+            // System.out.println("I m seeing if the storage works: " + storageForDeleted.peek().toString());
```
Comment (whipermr5) [inline src/seedu/addressbook/commands/RestoreCommand.java]: Same here.

File: src/seedu/addressbook/data/person/Address.java [hunk 4]
```diff
@@ -15,6 +15,8 @@
     public final String value;
     private boolean isPrivate;
 
+
+
```
Comment (whipermr5) [inline src/seedu/addressbook/data/person/Address.java]: Unrelated changes.

File: src/seedu/addressbook/parser/Parser.java [hunk 5]
```diff
@@ -11,17 +11,7 @@
...
-import seedu.addressbook.commands.Command;
-import seedu.addressbook.commands.DeleteCommand;
-import seedu.addressbook.commands.ExitCommand;
-import seedu.addressbook.commands.FindCommand;
-import seedu.addressbook.commands.HelpCommand;
-import seedu.addressbook.commands.IncorrectCommand;
-import seedu.addressbook.commands.ListCommand;
-import seedu.addressbook.commands.ViewAllCommand;
-import seedu.addressbook.commands.ViewCommand;
+import seedu.addressbook.commands.*;
```
Comment (whipermr5) [inline src/seedu/addressbook/parser/Parser.java]: Avoid star imports as this violates the [coding standard](https://oss-generic.github.io/process/codingStandards/CodingStandard-Java.html#package-and-import-statements).

File: src/seedu/addressbook/parser/Parser.java [hunk 6]
```diff
@@ -81,6 +71,10 @@ public Command parseCommand(String userInput) {
         case DeleteCommand.COMMAND_WORD:
             return prepareDelete(arguments);
 
+        //modification here:
```
Comment (whipermr5) [inline src/seedu/addressbook/parser/Parser.java]: Version control eliminates the need for this type of comment :)

File: src/seedu/addressbook/storage/StorageForDeleted.java [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+
+public class StorageForDeleted {
+    private static Stack<ReadOnlyPerson> storageForDeleted;
+
+    public StorageForDeleted() {
+        this.storageForDeleted = new Stack<ReadOnlyPerson>();
+    }
+
+    public void storeTheDeleted(ReadOnlyPerson personBeingDeleted) throws UniquePersonList.DuplicatePersonException {
+        // System.out.println("i m at storeTheDeleted");
```
Comment (whipermr5) [inline src/seedu/addressbook/storage/StorageForDeleted.java]: Be careful what you commit :)

File: src/seedu/addressbook/storage/StorageForDeleted.java [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+        storageForDeleted.add(0, personBeingDeleted);
+        // System.out.println(storageForDeleted.peek().toString());
```
Comment (whipermr5) [inline src/seedu/addressbook/storage/StorageForDeleted.java]: Same here.

File: src/seedu/addressbook/storage/StorageForDeleted.java [hunk 9]
```diff
@@ -0,0 +1,61 @@
...
+    }
+
+    public Person getPersonToBeRestored() {
+        try {
+            ReadOnlyPerson personBeingRestored = storageForDeleted.remove(0);
+            Person toAdd = (Person) personBeingRestored;
+            // System.out.println("i m done with casting");
```
Comment (whipermr5) [inline src/seedu/addressbook/storage/StorageForDeleted.java]: Same here.

File: test/java/seedu/addressbook/commands/FindCommandTest.java [hunk 10]
```diff
@@ -13,13 +13,14 @@
...
 import seedu.addressbook.data.person.ReadOnlyPerson;
+import seedu.addressbook.storage.StorageForDeleted;
 import seedu.addressbook.util.TypicalPersons;
 
 public class FindCommandTest {
 
     private final AddressBook addressBook = new TypicalPersons().getTypicalAddressBook();
     private final TypicalPersons td = new TypicalPersons();
-
+    private final StorageForDeleted storageForDeleted = new StorageForDeleted();
```
Comment (whipermr5) [inline test/java/seedu/addressbook/commands/FindCommandTest.java]: Keep the blank line before and add a blank line after.
